src/main/java/it/gov/pagopa/reward/service/synchronous/op/CreateTrxSynchronousServiceImpl.java

```diff
@@ -88,9 +90,10 @@ private Mono<Boolean> checkInitiative(SynchronousTransactionRequestDTO request,
                 .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_INITIATIVE_NOT_FOUND));
     }
 
-    private Mono<Boolean> checkOnboarded(SynchronousTransactionRequestDTO request, TransactionDTO trx, String initiativeId) {
+    private Mono<Pair<InitiativeConfig, OnboardingInfo>> checkOnboarded(SynchronousTransactionRequestDTO request, TransactionDTO trx, String initiativeId) {
         return onboardedInitiativesService.isOnboarded(trx.getHpan(), trx.getTrxChargeDate(), initiativeId)
-                .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE));
+                .switchIfEmpty(Mono.error(new TransactionSynchronousException(syncTrxRequest2TransactionDtoMapper
+                        .apply(request, initiativeId, List.of(RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE)))));
```

Comment:
you could re-use the checkingResult logic instead of write again the trxException create logic:

```suggestion
                .switchIfEmpty(Mono.error(checkingResult(false, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE)));
```